documented-derive/src/lib.rs

```diff
@@ -1,59 +1,113 @@
 use proc_macro::TokenStream;
 use quote::quote;
-use syn::{parse_macro_input, spanned::Spanned, DeriveInput, Error, Expr, ExprLit, Lit, Meta};
+use syn::{
+    parse_macro_input, parse_quote, spanned::Spanned, Attribute, Data, DataEnum, DataStruct,
+    DataUnion, DeriveInput, Error, Expr, ExprLit, Lit, Meta, Path,
+};
+
+fn crate_module_path() -> Path {
+    parse_quote!(::documented)
+}
 
 /// Derive proc-macro for `Documented` trait.
 #[proc_macro_derive(Documented)]
 pub fn documented(input: TokenStream) -> TokenStream {
     let input = parse_macro_input!(input as DeriveInput);
 
-    let ident = input.ident;
+    let doc_comments = match get_comment(&input.attrs) {
+        Ok(Some(r)) => r,
+        // Should we use an Option instead?
+        Ok(None) => "".to_string(),
+        Err(e) => return e.into_compile_error().into(),
+    };
```

Comment:
This is a real quandary huh.

I understand why you made it this way, otherwise using field docs would require the top-level to have docs too, which doesn't really make any sense. But it's not as expressive as I would like it. And making it into an `Option` creates a breaking change (also the UX would suck), which is undesirable too.

How about, we split these into two derive macros? One for top level and one for fields?

Reply:
I'll make some changes and you can take a look.